deepchem/feat/rdkit_grid_featurizer.py

```diff
@@ -0,0 +1,1266 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+__author__ = "Bharath Ramsundar and Evan Feinberg"
+__copyright__ = "Copyright 2016, Stanford University"
+__license__ = "GPL"
+
+import os
+import shutil
+import time
+import tempfile
+import hashlib
+import logging
+from rdkit import Chem
+from rdkit.Chem import AllChem
+import deepchem.utils.rdkit_util as rdkit_util
+
+import numpy as np
+from copy import deepcopy
+from collections import deque
+from functools import partial
+from deepchem.feat import ComplexFeaturizer
+from deepchem.utils.save import log
+"""
+http://stackoverflow.com/questions/38987/how-can-i-merge-two-python-dictionaries-in-a-single-expression
+"""
+
+
+def get_ligand_filetype(ligand_filename):
+  """Returns the filetype of ligand."""
+  if ".mol2" in ligand_filename:
+    return ".mol2"
+  elif ".sdf" in ligand_filename:
+    return "sdf"
+  elif ".pdbqt" in ligand_filename:
+    return ".pdbqt"
+  elif ".pdb" in ligand_filename:
+    return ".pdb"
+  else:
+    raise ValueError("Unrecognized_filename")
+
+
+def load_molecule(molecule_file, add_hydrogens=True, calc_charges=False):
+  return rdkit_util.load_molecule(molecule_file, add_hydrogens, calc_charges)
+
+
+def _merge_two_dicts(x, y):
+  """Given two dicts, merge them into a new dict as a shallow copy."""
+  z = x.copy()
+  z.update(y)
+  return z
+
+
+def _compute_centroid(coordinates):
```

Comment:
As a high level comment, this file has a number of functions of type `_name_of_function` for legacy reasons. Let's just go ahead and get rid of the first underscore to make them `name_of_function` (so `compute_centroid` here). A lot of these functions are general enough that we might as well expose them to users.